test/e2e/lib/helpers/broker_event_transformation_test_with_source.go

```diff
@@ -169,16 +169,16 @@ func BrokerEventTransformationMetricsTestHelper(client *lib.Client, projectID st
 			BrokerName:      brokerName,
 			BrokerNamespace: client.Namespace,
 			StartTime:       start,
-			CountPerTriggerWithRespCode: map[string]int64{
-				lib.TriggerNameWithRespCode(trigger.Name, http.StatusAccepted):     1,
-				lib.TriggerNameWithRespCode(respTrigger.Name, http.StatusAccepted): 1,
+			CountPerTrigger: map[lib.TriggerAssertionKey]int64{
+				// For metrics with response code.
+				{Name: trigger.Name, RespCode: http.StatusAccepted}:     1,
+				{Name: respTrigger.Name, RespCode: http.StatusAccepted}: 1,
 				// Metric from first two delivery attempts (which would fail).
-				lib.TriggerNameWithRespCode(trigger.Name, http.StatusBadRequest): 2,
-			},
-			CountPerTriggerNoRespCode: map[string]int64{
+				{Name: trigger.Name, RespCode: http.StatusBadRequest}: 2,
+
 				// For metrics without response code, we expect 3 trigger deliveries (first 2 from delivery failures).
-				trigger.Name:     3,
-				respTrigger.Name: 1,
+				{Name: trigger.Name}:     3,
+				{Name: respTrigger.Name}: 1,
```

Comment:
I'm not sure this will work with the maps combined unless the keys are filtered somehow. One possibility would be derive a no resp code map from the resp code map to use in the processing metrics. Another possibility would be to go back to having two separate maps as before.

Reply:
Another possibility might be to use some fancy cmp opts to compare only the relevant keys but that might be too fiddly.

Reply:
Ah you're right